core/common/src/main/java/alluxio/exception/PreconditionMessage.java

```diff
@@ -74,29 +70,14 @@
   // SEMICOLON! minimize merge conflicts by putting it on its own line
   ;
 
-  private final MessageFormat mMessage;
+  private final String mMessage;
 
   PreconditionMessage(String message) {
```

Comment:
Both option 2 and option 3 look good to me. Different from the conf strings, it is unnecessary to iterate over all the precondition messages. So I don't really think `enum` is a good fit.